docs/prod-email.md

```diff
@@ -0,0 +1,96 @@
+from servers is much easier than setting up outgoing email with an
+inbox product like Gmail.  If you for whatever reason attempt to use a
+Gmail account to send outgoing email, you will need to read this
+Google support answer and configure that account as
+["less secure"](https://support.google.com/accounts/answer/6010255);
+Gmail doesn't allow servers to send outgoing email by default.  Note
+also that the rate limits for Gmail are also quite low (e.g. 100 /
+day), so it's easy to get rate-limited.
+
+### Logging outgoing email to a file for prototyping
+
+If for prototyping, you don't want to bother setting up an email
+provider, you can add to `/etc/zulip/settings.py` the following:
+
+```
+EMAIL_BACKEND = 'django.core.mail.backends.filebased.EmailBackend'
+EMAIL_FILE_PATH = '/var/log/zulip/emails
+```
+
+Outgoing emails that Zulip would have sent will just be written to
+files in `/var/log/zulip/emails/`.  This is enough to get you through
+initial user registration without an SMTP provider.
+
+Remember to delete this configuration and restart the server if you
+later setup a real SMTP provider!
+
+### Configuration
+
+To configure outgoing SMTP, you will need to complete the following steps:
+
+1. Fill out the outgoing email sending configuration block in
+`/etc/zulip/settings.py`, including `EMAIL_HOST`, `EMAIL_HOST_USER`,
+`DEFAULT_FROM_EMAIL`, and `NOREPLY_EMAIL_ADDRESS`.  You may also need
+to set `EMAIL_PORT` if your provider doesn't use the standard
+SMTP submission port (587).
+
+2. Put the SMTP password for `EMAIL_HOST_USER` in
+`/etc/zulip/zulip-secrets.conf` as `email_password = yourPassword`.
+
+#### Testing and troubleshooting
+
+You can quickly test your outgoing email configuration using:
+
+```
+su zulip
+/home/zulip/deployments/current/manage.py send_test_email username@example.com
+```
+
+If it doesn't throw an error, it probably worked; you can confirm by
+checking your email.
+
+It's important to test, because outgoing email often doesn't work the
+first time.  Common causes of failures are:
+
+* Your hosting provider blocking outgoing SMTP traffic in its
+default firewall rules.  Check whether `EMAIL_PORT` is blocked in your
+hosting provider's firewall.
+* Forgetting to put the password in `/etc/zulip/zulip-secrets.conf`.
+* Typos in transcribing the username or password.
+
+Once you have it working from the management command, remember to
+restart your Zulip server using
+`/home/zulip/deployments/current/restart-server` so that the running
+server is using the latest configuration.
+
+#### Advanced troubleshooting
+
+With the exception of reading `EMAIL_HOST_PASSWORD` from the
+`email_password` entry in the Zulip secrets file so that secrets don't
+live in the `/etc/zulip/settings.py` file, Zulip uses Django's
+standard
+[SMTP backend](https://docs.djangoproject.com/en/1.10/topics/email/#smtp-backend),
+so if you're having issues, you may want to search for documentation
+on using your email provider with Django.
```

Comment:
Updated to:
```
Zulip's email sending configuration is based on the standard Django
[SMTP backend](https://docs.djangoproject.com/en/1.10/topics/email/#smtp-backend)
configuration.  The one thing we've changed from the defaults is
reading `EMAIL_HOST_PASSWORD` from the `email_password` entry in the
Zulip secrets file, so that secrets don't live in the
`/etc/zulip/settings.py` file.

So if you're having trouble getting your email provider working, you
may want to search for documentation related to using your email
provider with Django.
```